src/azure-cli/azure/cli/command_modules/storage/_validators.py

```diff
@@ -1577,6 +1577,55 @@ def validate_azcopy_credential(cmd, namespace):
                                             service=service, resource_types='sco', permissions='rl')
 
 
+def is_directory(props):
+    return 'hdi_isfolder' in props.metadata.keys() and props.metadata['hdi_isfolder'] == 'true'
+
+
+def validate_fs_directory_upload_destination_url(cmd, namespace):
+    destination = namespace.destination_path
+    container = namespace.destination_fs
+    kwargs = {'account_name': namespace.account_name,
+              'account_key': namespace.account_key,
+              'connection_string': namespace.connection_string,
+              'sas_token': namespace.sas_token}
+    client = blob_data_service_factory(cmd.cli_ctx, kwargs)
```

Comment:
could you use adls related client factory?

Reply:
Actually I was intended to at first. I need the client to a) query_properties b) make_url.
`DataLakeFileClient` have `get_file_properties` but no `make_file_url`.
`BlobService` have both interfaces so I chose the later

Reply:
Please use `DataLakeFileClient` with get_file_properties to get properties and use client.url to  return file url

Reply:
done